lib/platform-stats.js

```diff
@@ -199,3 +344,121 @@ export const updateTopMeasurementParticipants = async (createPgClient) => {
     Sentry.captureException(err)
   }
 }
+
+/**
+ * @param {import('pg').Client} pgClient
+ */
+export const aggregateAndCleanupRecentData = async (pgClient) => {
+  try {
+    await pgClient.query('BEGIN')
+
+    // The ON CONFLICT theoretically should not happen, because it's two days old data,
+    // which should not be updated anymore. But it's a good practice to have it.
+    await pgClient.query(`
+      WITH recent_station_details_summary AS (
+        SELECT
+          day,
+          SUM(accepted_measurement_count) AS accepted_measurement_count,
+          SUM(total_measurement_count) AS total_measurement_count,
+          COUNT(DISTINCT station_id) AS distinct_active_station_count,
+          COUNT(DISTINCT participant_id) AS distinct_participant_address_count
+        FROM
+          recent_station_details
+        WHERE
+          day <= CURRENT_DATE - INTERVAL '2 days'
+        GROUP BY
+          day
+      ),
+      recent_participant_subnets_summary AS (
+        SELECT
+          day,
+          COUNT(DISTINCT subnet) AS distinct_inet_group_count
+        FROM
+          recent_participant_subnets
+        WHERE
+          day <= CURRENT_DATE - INTERVAL '2 days'
+        GROUP BY
+          day
+      )
+      INSERT INTO daily_measurements_summary (
+        day,
+        accepted_measurement_count,
+        total_measurement_count,
+        distinct_active_station_count,
+        distinct_participant_address_count,
+        distinct_inet_group_count
+      )
+      SELECT
+        rsd.day,
+        rsd.accepted_measurement_count,
+        rsd.total_measurement_count,
+        rsd.distinct_active_station_count,
+        rsd.distinct_participant_address_count,
+        rps.distinct_inet_group_count
+      FROM
+        recent_station_details_summary rsd
+      JOIN
+        recent_participant_subnets_summary rps ON rsd.day = rps.day
+    `)
+
+    // Delete aggregated data from recent_station_details and recent_participant_subnets
+    await pgClient.query(`
+      DELETE FROM recent_station_details WHERE day <= CURRENT_DATE - INTERVAL '2 days'
+    `)
+    await pgClient.query(`
+      DELETE FROM recent_participant_subnets WHERE day <= CURRENT_DATE - INTERVAL '2 days'
+    `)
+
+    await pgClient.query('COMMIT')
+  } catch (err) {
+    console.error('Error aggregating and cleaning up recent data', err)
+    await pgClient.query('ROLLBACK')
+    Sentry.captureException(err)
+  }
```

Comment:
```suggestion
  } catch (err) {
    await pgClient.query('ROLLBACK')
    throw errr
  }
```

This lets the outer function handle the error and submit it to Sentry